package pb;


// TODO: Make each of our matrix terms into a different class,
// then call that. Ex:
// message DoubleMatrix {
//   required uint32 rows = 1;
//   required uint32 cols = 2;
//   repeated double data = 3 [packed=true];
// }

message BVP_params
{
    repeated double          start_param         =1 [packed=true];
    repeated double          goal_param          =2 [packed=true];
    required int32           row_count           =3;
    required int32           col_count           =4;
    repeated double          x_data              =5 [packed=true];
    repeated double          y_data              =6 [packed=true];
    repeated double          z_data              =7 [packed=true];
    required uint32           tau                 =8;
}

message BVP_policy
{
    repeated double          force               =1 [packed=true];
    repeated double          time                =2 [packed=true];
    repeated double          phi                 =3 [packed=true];
    repeated double          start_param         =4 [packed=true];
    repeated double          goal_param          =5 [packed=true];
    required int32           tau                 =6;
}

message BVP_check
{
    required bool            need                =1;
}

message Heightmap
{
    required int32           row_count           =1;
    required int32           col_count           =2;
    repeated double          x_data              =3 [packed=true];
    repeated double          y_data              =4 [packed=true];
    repeated double          z_data              =5 [packed=true];
}

message RegisterBVPReqMsg
{
    optional bytes          topic        	       =1;
    optional string         uri     		   =2;
}

message RegisterBVPRepMsg
{
    optional bool           success              = 1;
}
